eras/conway/impl/src/Cardano/Ledger/Conway/TxBody.hs

```diff
@@ -451,16 +456,45 @@ totalDRepDeposits pp txb = nDReps <×> depositPerDRep
     nDReps = getSum @Int (foldMap' (\case RegDRepTxCert {} -> 1; _ -> 0) (txb ^. certsTxBodyL))
     depositPerDRep = pp ^. ppDRepDepositL
 
+-- | Compute all the deposits in a TxBody
+--   This includes deposits for
+--   1) registering Stake
+--   2) registering a StakePool
+--   3) registering a DRep
+--   4) making a Proposal
+-- This is the contribution of a TxBody towards the total 'Obligations' of the system
+-- See the Tyeps and Functions Cardano.Ledger.CertState(Obligations,obligationCertState) for more information.
 totalTxDepositsConway ::
   Crypto c =>
   PParams (ConwayEra c) ->
   CertState (ConwayEra c) ->
   TxBody (ConwayEra c) ->
   Coin
 totalTxDepositsConway pp cs txb =
-  totalTxDepositsShelley pp cs txb
-    <> totalProposalDeposits pp txb
-    <> totalDRepDeposits pp txb
+  totalTxDepositsShelley pp cs txb -- Stake and StakePool
+    <> totalProposalDeposits pp txb -- only in Conway and later
+    <> totalDRepDeposits pp txb -- only in Conway and later
+
+drepRefundsConway :: ConwayEraTxBody era => CertState era -> TxBody era -> Coin
+drepRefundsConway certState txb = Coin (snd (foldl' accum (vsDReps (certVState certState), 0) (txb ^. certsTxBodyL)))
+  where
+    accum pair@(currMap, currRefund) (UnRegDRepTxCert cred (Coin n)) = case (Map.lookup cred currMap) of
+      (Just (DRepState _ _ (Coin m))) | n == m -> (Map.delete cred currMap, currRefund + m)
```

Comment:
This check is unnecessary, validity of deposits is checked in GOVCERT rule
```suggestion
      (Just (DRepState _ _ (Coin m))) -> (Map.delete cred currMap, currRefund + m)
```

Reply:
no longer applicable.